x/crosschain/keeper/msg_server_add_to_outtx_tracker.go

```diff
@@ -29,6 +29,19 @@ func (k msgServer) AddToOutTxTracker(goCtx context.Context, msg *types.MsgAddToO
 		return nil, observertypes.ErrSupportedChains
 	}
 
+	// the cctx must exist and be pending
+	cctx, err := k.CctxByNonce(ctx, &types.QueryGetCctxByNonceRequest{
+		ChainID: msg.ChainId,
+		Nonce:   msg.Nonce,
+	})
+	if err != nil || cctx == nil || cctx.CrossChainTx == nil {
+		return nil, cosmoserrors.Wrap(types.ErrCannotFindCctx, "cannot add out tx: no corresponding cctx found")
+	}
+	if !IsPending(*cctx.CrossChainTx) {
+		k.RemoveOutTxTracker(ctx, msg.ChainId, msg.Nonce)
```

Comment:
Is this to deal with situations where the outtxtracker is not removed when the outbound is finalized? 
Or if a tracker gets recreated , after it is finalized

Reply:
@kingpinXD I think this is for both/whatever cases (even if we don't know which happened and which not).

Reply:
I added some comment in [commit](https://github.com/zeta-chain/node/pull/1533/commits/e03c55c6eed9315ca7dd69b8ec387e465f838aa0)